implementation/src/main/java/io/smallrye/mutiny/Multi.java

```diff
@@ -310,6 +299,19 @@ static MultiCreateBy createBy() {
         return onItem().producePublisher(mapper).merge();
     }
 
+    /**
+     * Produces a new {@link Multi} invoking the given callback when an {@code item} event is fired by the upstream.
+     * <p>
+     * While this method allows implementing side-effects <em>sneakily</em>, it is recommended to not do so, and only use
+     * this method to execute side-effect free synchronous logic, like tracing.
```

Comment:
I really strongly object to this.

You're basically recommending that nobody ever use Hibernate Reactive here.

Reply:
Why?

Reply:
Because persistence is a side-effect.

Reply:
OK, talking a bit in the background.

Clement is not worried about side effects per se but rather by the use of a shared resource concurrently. That's why in the *general* case, this method could be problematic as you would apply changes on a shared resource where the sharing safety is not guaranteed by the programming model.

But JPA inherently imposes to users that they do not share sessions between multiple clients. It's not enforced (though Hibernate does detect it in some cases) but all assumptions are based on that. one session == one user and there is no concurrent updates of the same objects. All concurrency conflicts are resolved down at the DB level.

Anyways, let's have a conversation on the phone to clarify all that. @FroMage is interested in the conversation too as he has been using Mutiny in anger recently.

Reply:
OK.